src/observer.ts

```diff
@@ -181,10 +197,38 @@ export class DataLayerObserver {
       return;
     }
 
+    const monitors: Monitor[] = [];
+
     try {
       const handler = this.addHandler(source);
       handler.debug = !!debug;
 
+      try {
+        if (monitor) {
+          // use select to get the target with the desired properties
+          // NOTE using [()] to pick, etc returns a copy and not the actual data layer reference
```

Comment:
I feel like this is going to create a bit of mystery. I wonder how we could surface this during rule writing so that tag managers understand what's happening.